knora/dsplib/utils/onto_create_ontology.py

```diff
@@ -301,151 +303,189 @@ def create_ontology(input_file: str,
-                tmp = propclass["object"].split(':')
-                if len(tmp) > 1:
-                    if tmp[0]:
-                        object = propclass["object"]  # fully qualified name
+        prop_classes = ontology.get("properties")
+        if prop_classes:
+            for prop_class in prop_classes:
+                prop_name = prop_class.get("name")
+                prop_label = LangString(prop_class.get("labels"))
+                '''
+                get the super-property/ies if defined, valid forms are:
+                  - "prefix:super-property" : fully qualified name of property in another ontology. The prefix has to be 
+                    defined in the prefixes part.
+                  - "super-property" : super-property defined in the knora-api ontology
+                  - if omitted, "knora-api:hasValue" is assumed
+                '''
+                if prop_class.get("super"):
+                    super_props = list(map(lambda a: a if ':' in a else knora_api_prefix + a, prop_class.get("super")))
+                else:
+                    super_props = ["knora-api:hasValue"]
+                '''
+                get the "object" if defined, valid forms are:
+                  - "prefix:object_name" : fully qualified object. The prefix has to be defined in the prefixes part.
+                  - ":object_name" : The object is defined in the current ontology.
+                  - "object_name" : The object is defined in "knora-api"
+                '''
+                if prop_class.get("object"):
+                    tmp = prop_class.get("object").split(':')
+                    if len(tmp) > 1:
+                        if tmp[0]:
+                            prop_object = prop_class.get("object")  # fully qualified name
+                        else:
+                            prop_object = new_ontology.name + ':' + tmp[1]  # object refers to actual ontology
                     else:
-                        if verbose:
-                            newontology.print()
-                        object = newontology.name + ':' + tmp[1]
+                        prop_object = knora_api_prefix + prop_class.get("object")  # object refers to knora-api
                 else:
-                    object = "knora-api:" + propclass["object"]
-            else:
-                object = None
+                    prop_object = None
+                prop_subject = prop_class.get("subject")
+                gui_element = prop_class.get("gui_element")
+                gui_attributes = prop_class.get("gui_attributes")
+                if gui_attributes and gui_attributes.get("hlist"):
+                    gui_attributes["hlist"] = "<" + list_root_nodes[gui_attributes["hlist"]]["id"] + ">"
+                prop_comment = prop_class.get("comment")
+                if prop_comment:
+                    prop_comment = LangString(prop_comment)
+                else:
+                    prop_comment = "no comment given"
```

Comment:
does this mean, that each property that's created without a comment, gets the string `"no comment given"` added as a comment? if so, that seems a bit unelegant, right?